NebulaPatcher/Patches/Dynamic/PlanetModelingManager_Patch.cs

```diff
@@ -120,6 +120,16 @@ private static void InternalLoadPlanetsRequestGenerator(PlanetData[] planetsToLo
 
                 if (planetsToRequest.Any())
                 {
+                    // Make local planet load first
+                    int localplanetId = Multiplayer.Session.LocalPlayer.Data.LocalPlanetId;
+                    if (localplanetId == -1)
+                        localplanetId = UIVirtualStarmap_Transpiler.customBirthPlanet;
+
+                    if (planetsToRequest.Contains(localplanetId))
+                    {
+                        planetsToRequest.Remove(localplanetId);
+                        planetsToRequest.Insert(0, localplanetId);
+                    }
```

Comment:
```suggestion
                    if (planetsToRequest.Remove(localplanetId))
                    {
                        planetsToRequest.Insert(0, localplanetId);
                    }
```
avoid iterating through the list twice